http/rest/handler.go

```diff
@@ -0,0 +1,107 @@
+package rest
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"raccoon/config"
+	"raccoon/logger"
+	"raccoon/metrics"
+	"raccoon/pkg/collection"
+	"raccoon/pkg/deserialization"
+	"raccoon/pkg/identification"
+	pb "raccoon/pkg/proto"
+	"raccoon/pkg/serialization"
+	"time"
+)
+
+type Handler struct{}
+
+func (h *Handler) GetRESTAPIHandler(c collection.Collector) http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		contentType := r.Header.Get("Content-Type")
+		rw.Header().Set("Content-Type", contentType)
+		var group string
+		group = r.Header.Get(config.ServerWs.ConnGroupHeader)
+		if group == "" {
+			group = config.ServerWs.ConnGroupDefault
+		}
+		identifier := identification.Identifier{
+			ID:    r.Header.Get(config.ServerWs.ConnIDHeader),
+			Group: group,
+		}
+
+		d, s := h.getDeserializerSerializer(contentType)
+		res := &Response{}
+		if r.Body == nil {
+			metrics.Increment("batches_read_total", fmt.Sprintf("status=failed,reason=emptybody,conn_group=%s", identifier.Group))
+			logger.Errorf("[rest.GetRESTAPIHandler] %s no body", identifier)
```

Comment:
for all errors, should we include the group as well?

Reply:
@NNcrawler added this while adding the changes for the group logic, I thought it was better to make it consistent.